lib/liquid/tags/raw.rb

```diff
@@ -12,9 +12,10 @@ def initialize(tag_name, markup, parse_context)
     def parse(tokens)
       @body = ''
       while token = tokens.shift
-        if token =~ FullTokenPossiblyInvalid
+        if token =~ RawTagEnd
           @body << $1 if $1 != "".freeze
-          return if block_delimiter == $2
+          tokens.prepend($3)
```

Comment:
This is causing test failures when used with [liquid-c](https://github.com/shopify/liquid-c) which implements its own tokenizer (Liquid::C::Tokenizer) that doesn't implement prepend.  It might also be a bit awkward to implement Liquid::C::Tokenizer#prepend, because it parses tokens lazily so doesn't have a list to prepend to.

We will need to figure out an implementation that works with liquid-c.  It might be more natural for it to just implement Liquid::Raw.parse, in which case we could continue to prepend here.